pkg/common/client/config.go

```diff
@@ -38,52 +39,202 @@ func MergeSaramaSettings(config *sarama.Config, saramaSettingsYamlString string)
 		config.Version = constants.ConfigKafkaVersionDefault
 
 		// Add Any Required Settings
-		UpdateSaramaConfig(config, config.ClientID, "", "")
+		InitSaramaConfig(config)
```

Comment:
I'd really avoid having functions with strong side effects like this. Again, unless I know  InitSaramaConfig internals, I'd be surprised that I'm passing an object to a function so that it sets a couple of default settings on the object I'm passing. I think we should avoid this style of functions with heavy side effects. 

alternatives I can think of is setting the default settings explicitly,  or my least favorite, call it something that describes its intent (e.g `UpdateConfigWithDefaults` or `SetConfigDefaults` )

Reply:
I will go with `UpdateConfigWithDefaults` then.
Sounds good.